sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CompactibleFileStreamLog.scala

```diff
@@ -173,37 +177,64 @@ abstract class CompactibleFileStreamLog[T <: AnyRef : ClassTag](
   override def purge(thresholdBatchId: Long): Unit = throw new UnsupportedOperationException(
     s"Cannot purge as it might break internal state.")
 
+  /**
+   * Apply function on all entries in the specific batch. The method will throw
+   * FileNotFoundException if the metadata log file doesn't exist.
+   *
+   * NOTE: This doesn't fail early on corruption. The caller should handle the exception
```

Comment:
Shall we mention `IllegalStateException` explicitly because the caller should know what kind of exception it is?

Reply:
`IllegalStateException` is not the only exception this method can throw. It can also throw `IOException` as it deals with file, and also depends on `deserializeEntry`.

I think the type of exception won't affect what caller should do. Caller may want to deal with FNFE specifically, but still do the same for others, and even for FNFE it's safer to follow the guide on NOTE.